src/index.js

```diff
@@ -128,28 +139,121 @@ export default class ReactImgix extends Component {
+        break
+      case 'picture':
+        if (!component) {
+          _component = 'picture'
+        }
+
+        //
+        // we need to make sure an img is the last child so we look for one
+        //    in children
+        //    a. if we find one, move it to the last entry if it's not already there
+        //    b. if we don't find one, create one.
+
+        // make sure all of our children have key set, otherwise we get react warnings
+        _children = React.Children.map(React.Children.toArray(children),
+          (child, idx) => React.cloneElement(child, Object.assign({}, child.props, { key: buildKey(idx) }))
+        )
+
+        // look for an <img> or <ReactImgix type='img'> - at the bare minimum we
+        // have to have a single <img> element or else ie will not work.
+        let imgIdx = _children.findIndex(c =>
+          (c.type === 'img' || ((c.type.hasOwnProperty('name') && c.type.name === 'ReactImgix') && c.props.type === 'img'))
+        )
+
+        if (imgIdx === -1) {
+          // didn't find one or empty array - either way make a new component to
+          // put at the end. we pass in almost all of our props as defaults to
+          // our children, exceptions are:
+          //
+          //    bg - only <source> and <img> elements are allowable as children of
+          //         <picture> so we strip this option
+          //    children - we don't want to get recursive here
+          //    component - same reason as bg
+          //    type - specifically we're adding an img type so we hard-code this,
+          //           also letting type=picture through would infinitely loop
+
+          let imgProps = {
+            aggressiveLoad,
+            auto,
+            customParams,
+            crop,
+            entropy,
+            faces,
+            fit,
+            generateSrcSet,
+            src,
+            type: 'img',
+            ...other,
+            // make sure to set a unique key too
+            key: buildKey(_children.length + 1)
+          }
+
+          // we also remove className and styles if they exist - those passed in
+          // to our top-level component are set there, if you want them set on
+          // the child <img> element you can use `imgProps`.
+          delete imgProps.className
+          delete imgProps.styles
+
+          // have to strip out props set to undefined since they will override
+          // any defaultProps in the child
+          Object.keys(imgProps).forEach(k => {
+            if (imgProps[k] === undefined) delete imgProps[k]
+          })
 
-      childProps.src = _src
+          _children.push(React.createElement(this.constructor, imgProps))
+        } else if (imgIdx !== (_children.length - 1)) {
+          // found one, need to move it to the end
+          _children.splice(_children.length - 1, 0, _children.splice(imgIdx, 1)[0])
+        }
+        break
+      default:
+        // TODO - should we console.warn something here?
```

Comment:
I think `PropTypes` should handle the warning here - if they're ignoring that, they'd probably ignore our warning too.